File: SunnyFlamingo/ClientApp/src/app/models/goods/computerTechnologies/computerParts/CPUModel.ts
import { ComputerPartModel } from "../computerPartModel";

export class CPUModel extends ComputerPartModel{
  cpuSocketType: string;
  thermalDesignPower: number;
  numberOfCores: number;
  numberOfThreads: number;
  constructor(options: {
    name: string;
    price: number;
    isAvailable: string;
    description: string;
    producerId: string;
    manufacturerId: string;
    material: string;
    color: string;
    imageIds: string[];
    cpuSocketType: string;
    thermalDesignPower: number;
    numberOfCores: number;
    numberOfThreads: number;
  }){
    super(options);
    this.cpuSocketType = options.cpuSocketType;
    this.thermalDesignPower = +options.thermalDesignPower;
    this.numberOfCores = +options.numberOfCores;
    this.numberOfThreads = +options.numberOfThreads;
  }
}

File: SunnyFlamingo/Models/Goods/ComputerTechnologies/ComputerParts/CPUModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SunnyFlamingo.Models
{
    public class CPUModel: ComputerPartModel
    {
        public string CPUSocketType { get; set; }
        public int ThermalDesignPower { get; set; }
        public int NumberOfCores { get; set; }
        public int NumberOfThreads { get; set; }
    }
}
